azure smp nit-picks (#6465)
harness/developer-hub: docs/cloud-cost-management/get-started/ccm-smp/azure-smp.md

This topic walks you through the steps required to set up CCM for Azure in a self-managed platform.

**Figure: Azure CCM Self-Managed Enterprise Edition architecture diagram**
<DocImage path={require('./static/azure-smp-arch.png')} title="Click to view full size image" />

> The flow at a high level works very similar to our SaaS enviornment. We support syncing data from multiple Azure account exports residing in different source storage accounts into CCM via a destination storage account (staging).

Reviewer: ```suggestion
The flow at a high level works similarly to our SaaS environment. We support syncing data from multiple Azure account exports residing in different source storage accounts into CCM via a destination storage account (staging).
```

> The goal of this particular documentation page is to set up an Azure Destination Storage Account and Application Registration, then configue an existing Harness SMP deployment to utilize these components to act as a "staging area" for future Azure CURs to be processed and made available for our Harness CCM capabilities.

Reviewer 2: Azure Exports not CURs.

Reviewer 3: Azure Billing Exports.

> The goal of this particular documentation page is to set up an Azure Destination Storage Account and Application Registration, then configue an existing Harness SMP deployment to utilize these components to act as a "staging area" for future Azure CURs to be processed and made available for our Harness CCM capabilities.

Reviewer 2: Also we can move this maybe to the top of the page.

> The goal of this particular documentation page is to set up an Azure Destination Storage Account and Application Registration, then configue an existing Harness SMP deployment to utilize these components to act as a "staging area" for future Azure Billing Exports to be processed and made available for our Harness CCM capabilities.

Reviewer: ```suggestion
The goal of this particular documentation page is to set up an Azure Destination Storage Account and Application Registration, then configure an existing Harness Self-Managed Enterprise Edition deployment to utilize these components to act as a "staging area" for future Azure Billing Exports to be processed and made available for Harness CCM capabilities.
```

**Figure: Azure CCM Self-Managed Documentation Goal**
<DocImage path={require('./static/azure-smp-arch-initial-setup.jpeg')} title="Click to view full size image" />

> This goal will consist of four tasks. For Step 1, 2 and 3 please sign in to your [Azure Portal](https://portal.azure.com/#home): 

Reviewer: ```suggestion
This goal consists of four tasks. For Steps 1, 2, and 3 sign in to your [Azure Portal](https://portal.azure.com/#home): 
```

1. [Setup a new Application via App Registration](#setup-a-new-application-via-app-registration).
2. [Create a new Client secret](#create-a-new-client-secret).
3. [Create a new Destination Storage Account and a new Storage Container](#create-a-new-destination-storage-account-and-a-new-storage-container).
4. [Deploy workloads via Helm charts](#deploy-workloads-via-helm-charts).

:::info

> There are several times where we will ask you to note some items as suggested `variables`, please keep these values handy in a document or note of your choice as they will be used later in the process.

Reviewer: ```suggestion
There are several times when you'll need to note some items as suggested `variables`. Keep these values in a document or note as they will be used later in the process.
```

:::
   
## Setup a new Application via App Registration

Within your Microsoft Azure WebUI:
1. Navigate to [App registrations](https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationsListBlade).
2. Select "New Registration"
3. Give your App registration a Name like `Harness CCM App`.

> 4. For Supported account types select: `Accounts in any organizational directory (Any Microsoft Entra ID tenant - Multitenant)`.

Reviewer 4: Under "Supported account types", select:

> 5. Click on Register. For more information, go to [Quickstart: Register an App](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app).

Reviewer 4: "Register"

Reviewer 4: please make the options bold, whenever we are using them in our docs

7. Select **Expires** from the drop down select `730 days (24 months)`(or the maximum allowed time).
8. Click **Add** Button.

  <DocImage path={require('./static/azure-client-secret.png')} title="Click to view full size image" />

9. From the new Client secret, Note & Save **Value** → `clientSecret`. For more information, go to [Add credentials](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app#add-credentials).

:::info
Make a note of the following:

> - tenantId

Reviewer 4: please use backticks

13. Check all **Allowed resource types** which are `Service`, `Container` and `Object`.
14. Add 10 years to **End** in **Start and expiry date/time**.
15. Click on **Generate SAS and connection string**. This is required to sync data from multiple source account to destination storage account.
16. Note **SAS token** as → `sasToken` starting from `sv=`, ignore `?` in beginning. For more information, go to [Create your SAS tokens](https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/create-sas-tokens?view=doc-intel-4.0.0#use-azure-storage-explorer).

  <DocImage path={require('./static/azure-destination-storage-sas-token.png')} width="50%" height="50%" title="Click to view full size image" />

:::info
Make a note of the following:

> - storageName

Reviewer 4: backticks

- containerName
- sasToken
:::

## Deploy workloads via Helm charts
From here, we will use the values we've noted from previous steps to update our Harness SMP deployment.

> 1. Clone the chart repository if you have not already.

Reviewer 4: (if not done already)

## Handling Kubernetes secrets

When installing or upgrading the Helm charts, Kubernetes secrets with default values are created within the cluster. These generated secrets should be updated with the values mentioned above.

>  Before updating the secrets, you need to convert the secret into base64 encoded format. For example, if your **HARNESS_CE_AZURE_CLIENTID** value is "clientId", it would be stored as `Y2xpZW50SWQ==` after encoding. Please 

Reviewer 4: Base64

Reviewer 4: clientId should be in backticks

cleanupInterval = 0
```
</details>
After these secrets are implemented, please `kubectl delete` the `cloud-info` pod for this release to start a newly configured pod.

- nextgen-ce

```
kubectl edit secret nextgen-ce -n <namespace>
```

```
AZURE_APP_CLIENT_SECRET: "<base64 of clientSecret>"
```

> After these secrets are implemented, please `kubectl delete` the `nextgen` pods for this release to start a newly configured pod.

Reviewer 2: nextgen-ce pods

```

```
SMTP_HOST: <base64 of SMTP_HOST>
SMTP_PASSWORD: <base64 of SMTP_PASSWORD>
SMTP_PORT: <base64 of SMTP_PORT>
SMTP_USE_SSL: <base64 of SMTP_USE_SSL>
SMTP_USERNAME: <base64 of SMTP_USERNAME>
```

> After these secrets are implemented, please `kubectl delete` any pods related to `platform-service` for this release to start newly configured pods.

Reviewer 2: Also delete pods for batch-processing.

For your reference batch-processing also use SMTP secrets to send Perspective Reports:   https://github.com/harness/harness-core/blob/61d746ecac88cb82894e6c2f0d772e95dcdb3bef/batch-processing/chart/templates/statefulset.yaml#L65

:::info
Please ensure your Persistent Volumes related to TimescaleDB is at least 100Gi, run: `kubectl edit pvc wal-volume-harness-timescaledb-0 -n <namespace>`. These volumes are critical for capabilities like Recommendations and Anomalies within CCM.
:::

## Next steps

At this point, your "staging area" for future Microsoft Azure CUR reports is ready and we can follow our standard methodologies for onboarding these assets.

**Figure: Azure CCM Self-Managed Documentation Next Steps**
<DocImage path={require('./static/azure-smp-arch-next-steps.jpeg')} title="Click to view full size image" />

> - Use [Azure connector setup](https://developer.harness.io/docs/cloud-cost-management/get-started/onboarding-guide/set-up-cost-visibility-for-azure) to set up Azure CURs and push them to our configured destination storage account.

Reviewer 2: Azure Exports
